android/CONTRIBUTING_code.md

```diff
@@ -0,0 +1,81 @@
+# Contributing code to Mozilla's Android projects
+Thank you for taking the time to contribute to one of Mozilla's Android
+projects! 🔥 🦊 <3 🤖! 🎉 👍 For a full list of projects, see
+[the README](../README.md#android).
+
+Before contributing, please review our [Community Participation Guidelines].
+
+If you run into trouble at any point, ask for help! Check out our
+[preferred communication channels.](./CONTRIBUTING.md#communication)
+
+Contents:
+- [Finding issues to work on](#finding-issues-to-work-on)
+- [Creating a Pull Request](#creating-a-pull-request)
+- [Merging](#merging)
+
+## Finding issues to work on
+**New to Mozilla's mobile projects?** See issues labeled `good first issue` in your project's
+issues tracker (example: [focus-android][fa good first]). These are designed to be
+easier to implement so you can focus on learning our pull request workflow. *Please only
+fix one of these.*
+
+**Looking for more challenging issues?** See issues labeled `help wanted` (example:
+[focus-android][fa help]). These are issues that are ready to be implemented without
+additional product or UX discussion.
+
+**When you find an issue you'd like to work on,** *comment on the issue* saying that
+you'd like to work on it. This ensures it is still available for you to work on.
+
+**If you want to work on a new feature**, *always file an issue first* and wait
+for our team to discuss it. We want to ensure all teams (product, ux, engineering)
+have an opportunity to provide feedback. **Pull requests for unsolicited features
+are unlikely to get merged.**
+
+## Creating a Pull Request
+Our team follows [the GitHub pull request workflow][gh workflow]: fork, branch, commit,
+pull request, automated tests, review, merge. If you're new to GitHub, check out [the official
+guides][gh guides] for more information.
+
+An example commit message summary looks like, `Closes #5: Upgrade gradle to v1.3.0`.
+
+Please follow these guidelines for your pull requests:
+
+- All Pull Requests should address an issue. If your pull request doesn't have an
+issue, file it!
+  - GitHub search defaults to issues, not PRs, so ensuring there is an issue for your PR
+  means it'll be easier to find
+- The commit message summary should briefly describe what code changed in the commit, *not
+the issue you're fixing.*
+  - We encourage you to use the commit message body to elaborate what changed and why
+- Include the issue number in your commit messages. This links your PR to the issue it's
+intended to fix.
+  - If your PR closes an issue, include `Closes #...` in one of your commit messages. This
+  will automatically close the linked issue ([more info][auto close]).
+- Prefer "micro commits".
```

Comment:
Perhaps, on top of this, we should request their commits "tell a story." I'm concerned this may lead people to make way too many small commits that would ideally be a little more generic.

Reply:
Added a sub-bullet:

A series of micro commits should tell a story. For example, if your goal is to add a new
  icon to the toolbar, you can make a commit to add the icon asset and then make a commit to
  use the icon in the code.